### PR TITLE
Enforce name restriction for all the metrics

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: api/src/main/java/io/opentelemetry/metrics/Meter.java [hunk 1]
```diff
@@ -127,7 +128,8 @@
   /**
    * Returns a builder for a {@link GaugeDouble}.
    *
-   * @param name the name of the metric.
+   * @param name the name of the metric. Should be a ASCII string with a length no greater than 255
+   *     characters.
    * @return a {@code GaugeDouble.Builder}.
    * @throws NullPointerException if {@code name} is null.
    * @throws IllegalArgumentException if different metric with the same name already registered.
```
Comment (songy23) [inline api/src/main/java/io/opentelemetry/metrics/Meter.java]: Nit: update this with illegal metric names.

File: api/src/main/java/io/opentelemetry/metrics/Meter.java [hunk 2]
```diff
@@ -138,7 +140,8 @@
   /**
    * Returns a builder for a {@link CounterDouble}.
    *
-   * @param name the name of the metric.
+   * @param name the name of the metric. Should be a ASCII string with a length no greater than 255
+   *     characters.
    * @return a {@code CounterDouble.Builder}.
    * @throws NullPointerException if {@code name} is null.
    * @throws IllegalArgumentException if different metric with the same name already registered.
```
Comment (songy23) [inline api/src/main/java/io/opentelemetry/metrics/Meter.java]: ditto

Comment (bogdandrutu) [reply api/src/main/java/io/opentelemetry/metrics/Meter.java]: Done.

File: api/src/main/java/io/opentelemetry/metrics/Meter.java (hunk 1 above)
Comment (bogdandrutu) [reply api/src/main/java/io/opentelemetry/metrics/Meter.java]: Done.

**Pull request merged**